### DISCUSSION TITLE
Environment variables in the GitHub app

### DISCUSSION THREAD

**Discussion by Bocom** [renovatebot/renovate#9147] [Request Help]
Hi,

I have a Composer dependency ([`pivvenit/acf-pro-installer`](https://github.com/pivvenit/acf-pro-installer)) that expects either an environment variable (read either from the environment or a `.env` file) named `ACF_PRO_KEY`, or a key named `acf-pro-key` in the global Composer `config.json` file. I would obviously prefer to not commit this key to the repository in a `.env` file.

Is there a way to work with these limitations in the GitHub app, or is the self-hosted version the only way to go with this?

**Comment by viceice** [comment]
You need to use the selfhostet version with [customEnvVariables](https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables)

**Reply by rarkins** [reply]
We *are* looking at allowing custom env variables configured by users, but need to fully evaluate the security implications first. There are certain env variables you can specify which can be used to compromise a host etc.
